### DISCUSSION TITLE
Tailwind Integration

### DISCUSSION THREAD

**Discussion by mcDevnagh** [rustshop/htmx-sorta#11] [General]
>  Unfortunately it doesn't have a great Rust/maud integration, so I use `twind` as a client-side Tailwind JIT instead.

https://github.com/rustshop/htmx-sorta/blob/master/README.md#about-the-stack

Tailwind works! The following `tailwind.config.js` works with the Tailwind CLI, since it will read the Rust source code and find the class names.
```js
/** @type {import('tailwindcss').Config} */
module.exports = {
	content: ["./src/**/*.rs"],
	theme: {
		extend: {},
	},
	plugins: [],
}
```

Also, this is a high-ranking link in DuckDuckGo when searching a few different combinations of `tailwind maud rust htmx`, so updating the [README](https://github.com/rustshop/htmx-sorta/blob/master/README.md) would do a lot of help for devs. It would've saved me a lot of time experimenting with twind, when tailwind works even better than twind. I only stopped experimenting with twind because I ran into https://github.com/tw-in-js/twind/issues/453

**Comment by dpc** [comment]
Thanks for letting me know! Updated the PR in #12 ! I'll try it out next time I have some time to play with it.

**Comment by dpc** [comment]
I've switched to this approach in https://github.com/rustshop/htmx-sorta/pull/14

**Reply by dpc** [reply]
I wonder if I should make style.css be compiled with build.rs...

**Reply by mcDevnagh** [reply]
I didn't even know [that](https://doc.rust-lang.org/cargo/reference/build-scripts.html) existed
That would be very cool. I use an npm script, since I installed it via npm, and it's in the local node_modules, but I wonder if `build.rs` can leverage [the rust implementation](https://docs.rs/tailwind-css/latest/tailwind_css/)

**Reply by dpc** [reply]
Possibly, I don't know how feature complete it is.

Calling normal `tailwind-css` takes around 300ms on my system, which seems a bit too slow to do on every `cargo build`. Hopefully Rust version is much faster.
